cluster-operator/src/main/java/io/strimzi/operator/cluster/model/nodepools/NodePoolUtils.java

```diff
@@ -198,4 +211,140 @@ private static List<String> validateKRaftProcessRoles(List<KafkaNodePool> nodePo
 
         return errors;
     }
+
+    /**
+     * Extracts the cluster IDs found in the Kafka and KafkaNodePool custom resources and validates whether the
+     * resources do not belong to different Kafka cluster. This method is used for ZooKeeper based Kafka clusters. In
+     * ZooKeeper based clusters, Kafka generates the cluster ID on its own. So the Node Pools cannot have a cluster ID
+     * before the Kafka cluster.
+     *
+     * @param reconciliation    Reconciliation marker
+     * @param kafka             The Kafka custom resource which should be validates
+     * @param pools             The list of KafkaNodePool custom resources which should be validated
+     *
+     * @return  List with errors found while checking the cluster IDs or null if no issues found
+     */
+    public static List<String> validateZooKeeperBasedClusterIds(Reconciliation reconciliation, Kafka kafka, List<KafkaNodePool> pools)   {
+        String kafkaCrClusterId = null; // Stores the cluster ID found in the Kafka CR
+        boolean hasClusterIdTooEarly = false; // indicates if some node pool has a cluster ID before the Kafka CR
+        boolean hasConflict = false; // indicates if multiple different cluster IDs have been found
+
+        // Try to extract the cluster ID from the Kafka CR status first
+        if (kafka.getStatus() != null && kafka.getStatus().getClusterId() != null) {
+            kafkaCrClusterId = kafka.getStatus().getClusterId();
+        }
+
+        // Go through the pools, try to extract the Cluster ID and validate it against already found
+        // For ZooKeeper based clusters, the Node Pools are allowed to have:
+        //     * No cluster ID regardless whether the Kafka CR has some cluster ID or not (e.g. with new node pools)
+        //     * The same cluster ID as the Kafka CR if the Kafka CR has a cluster ID already
+        //
+        // But they are not allowed to have any cluster ID when the Kafka CR does not have one yet.
+        for (KafkaNodePool pool : pools)   {
+            if (pool.getStatus() != null && pool.getStatus().getClusterId() != null)   {
+                if (kafkaCrClusterId == null)   {
+                    hasClusterIdTooEarly = true;
+                    LOGGER.warnCr(reconciliation, "Cluster ID {} found in KafkaNodePool {} is not allowed without the same cluster ID being present in the Kafka custom resource", pool.getStatus().getClusterId(), pool.getMetadata().getName());
```

Comment:
I would add the name of the Kafka CR as well, to have a single warning line to understand from which CRs the cluster-ids are got and compared.

Reply:
@ppatierno I'm not sure that is needed. It has the name of the Kafka CR in the reconciliation marker. WDYT?

Reply:
Ah right, ok no need to add more.